### DISCUSSION TITLE
Series Analysis Question

### DISCUSSION THREAD

**Discussion by johnlwagner** [dtcenter/METplus#2119] [Statistical Computation] [answered]
Greetings
I have a set of netcdf files produced by Grid Stat that contain the LATLON and DIFF fields, and not the RAW (fcst and obs) fields.  I would like to compute the MAE, ME, and RMSE using these DIFF fields in Series Analysis.  If this is possible, can you please point me to the documentation on how to set up the Series Analysis Config file for this.  I have been unable to locate how to do this so far.
Thanks
John

**Comment by CPKalb** [comment]
Hi John:

Are you using METplus or MET directly?  The setup will be slightly different depending on which way you are running MET.

If you are running MET directly, you would need to set your diff fields as the input variable names and levels for both the fcst and obs variables.  If you post the header from ncdump on one of the netcdf files, I could give more detailed information on how to set that up.  Then, to get MAE, ME, and RMSE, set the following:

output_stats = {
   fho    = [];
   ctc    = [];
   cts    = [];
   mctc   = [];
   mcts   = [];
   cnt    = ["MAE", "ME", "RMSE"];
   sl1l2  = [];
   sal1l2 = [];
   pct    = [];
   pstd   = [];
   pjc    = [];
   prc    = [];
}

To run MET, you would then want to do the following:
series_analysis -both file1..file_n -out output_file -config config_file

If you are using METplus, we don't have an example using output from grid-stat.  But, the basic use case has many of the settings.  In this case, you would want to set BOTH_SERIES_ANALYSIS_INPUT_DIR and BOTH_SERIES_ANALYSIS_INPUT_TEMPLATE, rather than setting them separately for the fcst and obs.  Then, instead of SERIES_ANALYSIS_OUTPUT_STATS_CNT = TOTAL, RMSE, FBAR, OBAR, set SERIES_ANALYSIS_OUTPUT_STATS_CNT = MAE, ME, RMSE

Let me know if this helps.

Thanks,
Christina

**Comment by johnlwagner** [comment]
Thanks Christina.  I am using MET directly.  Here is what the fcst/obs part
of my series analysis config file looks like currently (FIELD=wd,
LEVEL=Z10).  This is the part that I think I am getting wrong.
//
// Forecast and observation fields to be verified
//
name  = "DIFF_${FIELD}_FULL";
fcst = {

   field = [
      {
        level = "${LEVEL}";
      }
   ];
}
obs = {
   field = [
      {
        level = "${LEVEL}";
      }
   ];
}

Here is an ncdump -h of one of my input files:
netcdf ndfd_grid_stat_120000L_20230201_000000V_pairs {
dimensions:
        lat = 1597 ;
        lon = 2345 ;
variables:
        float lat(lat, lon) ;
                lat:long_name = "latitude" ;
                lat:units = "degrees_north" ;
                lat:standard_name = "latitude" ;
        float lon(lat, lon) ;
                lon:long_name = "longitude" ;
                lon:units = "degrees_east" ;
                lon:standard_name = "longitude" ;
        float DIFF_wd_FULL(lat, lon) ;
                DIFF_wd_FULL:name = "DIFF_wd_FULL" ;
                DIFF_wd_FULL:long_name = "Wind Direction (from which
blowing) at Z10 and Wind Direction (from which blowing) at Z10" ;
                DIFF_wd_FULL:level = "Z10 and Z10" ;
                DIFF_wd_FULL:units = "degtrue and degtrue" ;
                DIFF_wd_FULL:init_time = "20230131_120000" ;
                DIFF_wd_FULL:init_time_ut = "1675166400" ;
                DIFF_wd_FULL:valid_time = "20230201_000000" ;
                DIFF_wd_FULL:valid_time_ut = "1675209600" ;
                DIFF_wd_FULL:accum_time = "NA" ;
                DIFF_wd_FULL:accum_time_sec = -9999 ;
                DIFF_wd_FULL:_FillValue = -9999.f ;
                DIFF_wd_FULL:desc = "NA" ;
                DIFF_wd_FULL:masking_region = "FULL" ;
                DIFF_wd_FULL:smoothing_method = "NEAREST" ;
                DIFF_wd_FULL:smoothing_neighborhood = 1 ;

// global attributes:
                :FileOrigins = "File
/lfs/h1/mdl/ptmp/john.l.wagner/verif/test1/ndfd_series_analysis_urma_ndfd_stats_20230200_co_wd_grid_valid.153568/NDFDproj/012/grid_stat_120000L_20230201_000000V_pairs.nc
generated 20230412_142051 UTC on host nid002252 by the MET grid_stat tool" ;
                :MET_version = "V10.0.1" ;
                :MET_tool = "grid_stat" ;
                :model = "NDFD" ;
                :obtype = "URMA" ;
                :Difference = "Forecast Value - Observation Value" ;
                :Projection = "Lambert Conformal" ;
                :hemisphere = "N" ;
                :scale_lat_1 = "25.000000" ;
                :scale_lat_2 = "25.000000" ;
                :lat_pin = "19.228976" ;
                :lon_pin = "-126.276552" ;
                :x_pin = "0.000000" ;
                :y_pin = "0.000000" ;
                :lon_orient = "-95.000000" ;
                :d_km = "2.539703" ;
                :r_km = "6371.200000" ;
                :nx = "2345" ;
                :ny = "1597 grid_points" ;
}
Everything I have run so far gives me a map of all zeros, indicating that
the DIFF field is being used for both the fcst and obs.
[image: series_cnt_MAE_in_sera_ndfd_co_202302_12_wd_12_valid_test.png]
If you have access to dogwood, here is my test script:
/lfs/h1/mdl/ptmp/john.l.wagner/verif/test1/ndfd_series_analysis_urma_ndfd_stats_20230200_co_wd_grid_valid.227737/NDFDproj/012/test_sera.sh

Thanks
John


On Thu, Apr 13, 2023 at 10:42 PM Christina Kalb ***@***.***>
wrote:

> Hi John:
>
> Are you using METplus or MET directly? The setup will be slightly
> different depending on which way you are running MET.
>
> If you are running MET directly, you would need to set your diff fields as
> the input variable names and levels for both the fcst and obs variables. If
> you post the header from ncdump on one of the netcdf files, I could give
> more detailed information on how to set that up. Then, to get MAE, ME, and
> RMSE, set the following:
>
> output_stats = {
> fho = [];
> ctc = [];
> cts = [];
> mctc = [];
> mcts = [];
> cnt = ["MAE", "ME", "RMSE"];
> sl1l2 = [];
> sal1l2 = [];
> pct = [];
> pstd = [];
> pjc = [];
> prc = [];
> }
>
> To run MET, you would then want to do the following:
> series_analysis -both file1..file_n -out output_file -config config_file
>
> If you are using METplus, we don't have an example using output from
> grid-stat. But, the basic use case has many of the settings. In this case,
> you would want to set BOTH_SERIES_ANALYSIS_INPUT_DIR and
> BOTH_SERIES_ANALYSIS_INPUT_TEMPLATE, rather than setting them separately
> for the fcst and obs. Then, instead of SERIES_ANALYSIS_OUTPUT_STATS_CNT =
> TOTAL, RMSE, FBAR, OBAR, set SERIES_ANALYSIS_OUTPUT_STATS_CNT = MAE, ME,
> RMSE
>
> Let me know if this helps.
>
> Thanks,
> Christina
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/dtcenter/METplus/discussions/2119#discussioncomment-5611085>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGZ7TYP6BXWDBW5NXLKA6Z3XBC2PLANCNFSM6AAAAAAW376JUU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
John Wagner
Verification Task Lead
NOAA/National Weather Service
Meteorological Development Laboratory
Digital Forecast Services Division
SSMC2 Room 10106
Silver Spring, MD 20910
(240) 622-9138 <https://mail.google.com/mail/u/0/#m_226873644979562856_>
(Google
Voice)
(301) 427-9471 (office)
(908) 902-4155 (cell/text)

**Reply by CPKalb** [reply]
Hi John:

In your configuration listed, the diff field is indeed being used for both the forecast and obs.  This is because the variable "name" is outside the forecast and observation dictionaries.  To use separate fields for both the forecast and observation, you would want to put this field inside each respective dictionary.  Additionally, MET levels for netCDF files are specified differently than grib files.  In this case, you likely need "(*,*)" for the level.   This would make your dictionary 

fcst = {

   field = [
      {
        name = "DIFF_wd_FULL";
        level = "(*,*)";
      }
   ];
}
obs = {
   field = [
      {
       name = "obs_var_name";
        level = "obs_var_level";
      }
   ];
}

It sounds like you are wanting to use the diff field as the forecast.  But, what do you want to use for your observation field?

Christina

**Comment by johnlwagner** [comment]
The DIFF field is already the difference between the fcst and obs from grid
stat.  I was hoping just to aggregate them in series_analysis and compute
ME/MAE/RMSE instead of having series analysis compute the differences from
the fcst and obs, if that is possible.

On Fri, Apr 14, 2023 at 12:58 PM Christina Kalb ***@***.***>
wrote:

> Hi John:
>
> In your configuration listed, the diff field is indeed being used for both
> the forecast and obs. This is because the variable "name" is outside the
> forecast and observation dictionaries. To use separate fields for both the
> forecast and observation, you would want to put this field inside each
> respective dictionary. Additionally, MET levels for netCDF files are
> specified differently than grib files. In this case, you likely need "(*,*)"
> for the level. This would make your dictionary
>
> fcst = {
>
> field = [
> {
> name = "DIFF_wd_FULL";
> level = "(*,*)";
> }
> ];
> }
> obs = {
> field = [
> {
> name = "obs_var_name";
> level = "obs_var_level";
> }
> ];
> }
>
> It sounds like you are wanting to use the diff field as the forecast. But,
> what do you want to use for your observation field?
>
> Christina
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/dtcenter/METplus/discussions/2119#discussioncomment-5618413>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGZ7TYJ5LAIXTSV7ONLBLIDXBF627ANCNFSM6AAAAAAW376JUU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
John Wagner
Verification Task Lead
NOAA/National Weather Service
Meteorological Development Laboratory
Digital Forecast Services Division
SSMC2 Room 10106
Silver Spring, MD 20910
(240) 622-9138 <https://mail.google.com/mail/u/0/#m_226873644979562856_>
(Google
Voice)
(301) 427-9471 (office)
(908) 902-4155 (cell/text)

**Reply by CPKalb** [reply, marked as answer]
Hi John:

I am not sure that I fully understand what you are wanting to do. To compute ME/MAE/RMSE in MET, you need a forecast and observation field, which will then compute statistics using the [formulas](https://met.readthedocs.io/en/latest/Users_Guide/appendixC.html#mean-squared-error-mse).  There isn't a way to compute them using only the difference field.

Depending on how you want to aggregate, Stat-Analysis might do what you want, as ME/MAE/RMSE can be aggregated in Stat-Analysis from the SL1L2 output line.

Christina

**Comment by johnlwagner** [comment]
Thanks for confirming Christina.  Based on the documentation, this is what
I thought.
If I want gridpoint verification, I can't use grid stat to preprocess any
of the daily data, other than to create the matched pairs file.  I must use
series analysis to compute the daily differences and to aggregate the
scores.
Thanks
John

On Fri, Apr 14, 2023 at 1:30 PM Christina Kalb ***@***.***>
wrote:

> Hi John:
>
> I am not sure that I fully understand what you are wanting to do. To
> compute ME/MAE/RMSE in MET, you need a forecast and observation field,
> which will then compute statistics using the formulas
> <https://met.readthedocs.io/en/latest/Users_Guide/appendixC.html#mean-squared-error-mse>.
> There isn't a way to compute them using only the difference field.
>
> Depending on how you want to aggregate, Stat-Analysis might do what you
> want, as ME/MAE/RMSE can be aggregated in Stat-Analysis from the SL1L2
> output line.
>
> Christina
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/dtcenter/METplus/discussions/2119#discussioncomment-5618764>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGZ7TYLQLIJ77DSGFBTFPCTXBGCUJANCNFSM6AAAAAAW376JUU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
John Wagner
Verification Task Lead
NOAA/National Weather Service
Meteorological Development Laboratory
Digital Forecast Services Division
SSMC2 Room 10106
Silver Spring, MD 20910
(240) 622-9138 <https://mail.google.com/mail/u/0/#m_226873644979562856_>
(Google
Voice)
(301) 427-9471 (office)
(908) 902-4155 (cell/text)

**Reply by CPKalb** [reply]
Hi John:

Yes, that is correct.

Christina
